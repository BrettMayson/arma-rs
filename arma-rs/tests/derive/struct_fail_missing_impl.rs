use arma_rs_proc::{FromArma, IntoArma};

struct MyType;

#[derive(IntoArma, FromArma)]
struct DeriveTest(MyType);

#[derive(IntoArma, FromArma)]
struct Unit<T>(T);

fn main() {}
