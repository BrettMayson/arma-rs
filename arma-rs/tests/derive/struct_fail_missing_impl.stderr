error[E0599]: no method named `to_arma` found for struct `MyType` in the current scope
 --> tests/derive/struct_fail_missing_impl.rs:5:10
  |
3 | struct MyType;
  | ------------- method `to_arma` not found for this struct
4 |
5 | #[derive(IntoArma, FromArma)]
  |          ^^^^^^^^ method not found in `MyType`
  |
  = help: items from traits can only be used if the trait is implemented and in scope
  = note: the following trait defines an item `to_arma`, perhaps you need to implement it:
          candidate #1: `IntoArma`
  = note: this error originates in the derive macro `IntoArma` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `MyType: FromArma` is not satisfied
 --> tests/derive/struct_fail_missing_impl.rs:5:20
  |
5 | #[derive(IntoArma, FromArma)]
  |                    ^^^^^^^^ the trait `FromArma` is not implemented for `MyType`
  |
  = help: the following other types implement trait `FromArma`:
            (A, B)
            (A, B, C)
            (A, B, C, D)
            (A, B, C, D, E)
            (A, B, C, D, E, F)
            (A, B, C, D, E, F, G)
            (A, B, C, D, E, F, G, H)
            (A, B, C, D, E, F, G, H, I)
          and $N others
  = note: this error originates in the derive macro `FromArma` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `to_arma` found for type parameter `T` in the current scope
 --> tests/derive/struct_fail_missing_impl.rs:8:10
  |
8 | #[derive(IntoArma, FromArma)]
  |          ^^^^^^^^ method not found in `T`
9 | struct Generic<T>(T);
  |                - method `to_arma` not found for this type parameter
  |
  = help: items from traits can only be used if the type parameter is bounded by the trait
  = note: this error originates in the derive macro `IntoArma` (in Nightly builds, run with -Z macro-backtrace for more info)
help: the following trait defines an item `to_arma`, perhaps you need to restrict type parameter `T` with it:
  |
9 | struct Generic<T: IntoArma>(T);
  |                 ++++++++++

error[E0277]: the trait bound `T: FromArma` is not satisfied
 --> tests/derive/struct_fail_missing_impl.rs:8:20
  |
8 | #[derive(IntoArma, FromArma)]
  |                    ^^^^^^^^ the trait `FromArma` is not implemented for `T`
  |
  = note: this error originates in the derive macro `FromArma` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider restricting type parameter `T`
  |
9 | struct Generic<T: arma_rs::FromArma>(T);
  |                 +++++++++++++++++++
